metaflow/plugins/kfp/kfp.py

```diff
@@ -413,21 +443,75 @@ def _set_container_settings(container_op: ContainerOp, kfp_component: KfpCompone
                 vendor=resource_requirements["gpu_vendor"],
             )
 
-    def step_op(self, step_name: str) -> Callable[..., ContainerOp]:
+    def step_op(
+        self,
+        step_name: str,
+        kfp_component_inputs: List[str] = None,
+        kfp_component_outputs: List[str] = None,
+    ) -> Callable[..., ContainerOp]:
         """
         Workaround of KFP.components.func_to_container_op() to set KFP Component name
         """
         # KFP Component for a step defined in the Metaflow FlowSpec.
         step_op_component: Dict = yaml.load(
             kfp.components.func_to_component_text(
-                kfp_step_function, base_image=self.base_image
+                KubeflowPipelines._update_step_op_func_signature(
+                    kfp_step_function,
+                    kfp_component_inputs=kfp_component_inputs,
+                    kfp_component_outputs=kfp_component_outputs,
+                ),
+                base_image=self.base_image,
             ),
             yaml.SafeLoader,
         )
 
         step_op_component["name"] = step_name
         return kfp.components.load_component_from_text(yaml.dump(step_op_component))
 
+    @staticmethod
+    def _update_step_op_func_signature(
+        func: Callable,
+        kfp_component_inputs: List[str],
+        kfp_component_outputs: List[str],
+    ) -> Callable:
```

Comment:
Do you mind adding in a docstring for this method. Obviously seems quite important so might be good to have a short description of what it's doing to ingest it at a glance :)